### PR TITLE
Update games to limit viewing users to respective games

### PR DESCRIPTION
**Pull request by FifthSurprise**
This PR Updates the administration so that non-superamdin game control can edit/maintain users.

The users are limit to users assigned to that game.

This is a temporary placeholder as I need to get stubbing out users in Devise so that I can test controller work.

### CODE REVIEW

**Comment by FifthSurprise** [comment]
And added the tests. @michaelpelletier @gsaint 
Will merge in a few days unless objections.

File: app/views/users/edit.html.erb [hunk 1]
```diff
@@ -30,9 +30,10 @@
       <%= f.label :team_role %> </br>
         <%= f.select :team_role, options_for_select(@team_roles.collect{ |role| [role.role_display_name, role.id]}, selected = @user.team_role_id) %></br><br />
 
+      <% if @super_admin %>
       <%= f.label :game %> </br>
```
Comment (michaelpelletier) [inline app/views/users/edit.html.erb]: Indenting looks like it might be off. could just be github.

File: app/views/users/new.html.erb [hunk 2]
```diff
@@ -31,9 +31,11 @@
       <%= f.label :team_role %> </br>
         <%= f.select :team_role, options_for_select(@team_roles.collect{ |role| [role.role_display_name, role.id]}) %></br><br />
 
+      <% if @super_admin? %>
```
Comment (michaelpelletier) [inline app/views/users/new.html.erb]: Indenting as above.

File: spec/controllers/users_controller_spec.rb [hunk 3]
```diff
@@ -1,5 +1,44 @@
...
+      unowned_game.users << user2
+
+      allow(controller).to receive(:current_user).and_return(control)
+      get :index
+
+      expect(controller.instance_variable_get(:@users)).to include(control, user1)
+      expect(controller.instance_variable_get(:@users)).not_to include(user2)
+    end
+
+    it 'shows all users tied to your game if you are a super admin' do
```
Comment (gsaint) [inline spec/controllers/users_controller_spec.rb]: Shall this be `shows all users if you are a super admin`?

Comment (FifthSurprise) [inline spec/controllers/users_controller_spec.rb]: yeah that makes more sense.

**Pull request merged**